site-src/geps/gep-1324.md

```diff
@@ -0,0 +1,127 @@
+
+### **Service**
+
+"service" is a key component to meshes. In general mesh parlance, the term “service” represents an arbitrary cluster of related workloads accessible via the network as a single entity. In a Kubernetes world, this compute can be a typical `Service`, but it can also be something like a `ServiceImport` (from the MCS spec), or a custom type (a made up example would be a `CloudService` abstracting cloud endpoints). Due to the naming overlap with the Kubernetes `Service` resource, it becomes necessary to use more precise terms when describing these concerns, reserving `Service` exclusively for the Kubernetes resource.
+
+While a `Service` in Kubernetes is a single resource, logically it handles two different roles - a "**service frontend**" and "**service backend**".
+
+<div float="left">
+  <figure style="display:inline-block;float:left;text-align: center; width: 50%;">
+    <img src="../images/1324-resource-view-of-service.png"/>
+    <figcaption>Resource level view of Service</figcaption>
+  </figure>
+  <figure style="display:inline-block;float:left;text-align: center; width: 50%;">
+    <img src="../images/1324-decomposed-view-of-service.png"/>
+    <figcaption>Decomposed view of Service</figcaption>
+  </figure>
+</div>
+<br />
+
+The "service frontend" refers to how we call a service. In a `Service`, this is an automatically allocated DNS name (`name.namespace.svc.cluster.local`) and IP address (`ClusterIP`). Not all services have frontends - for example, "headless" `Service`s.
+
+The "service backend" refers to the target of a service. In `Service`, this is `Endpoints` (or `EndpointSlice`).
+
+In the Gateway API today, `Service` is used as a "service backend". For example, in a route we would refer to `Service` as the (aptly named) `backendRef`:
+
+![image displaying how a route uses the backend function of a service](images/1324-backend-ref.png "image displaying how a route uses the backend function of the Service resource")
+
+A user connecting to a Gateway, which has a Route forwarding to "Service Backend"
+
+However, there are other ways that services _could_ be used in the API by utilizing the "service frontend".
+
+Below shows a hypothetical way to model splitting calls to the `a` service frontend to backends composed of `a-v1` and `a-v2` (see [real world example](https://linkerd.io/2.11/tasks/canary-release/)). In this case, service is used as both a frontend and a backend. The route attaches to "Service a frontend", and any calls to that will be directed to the `a-v1` and `a-v2` backends (as determined by the logic within the route.
+
+![image showing how a route could use the frontend function of the Service resource](images/1324-service-frontend.png "image showing how a route could use the frontend function of the Service resource")
+
+A user connecting to a service (frontend), which has a Route forwarding to two different services (backends)
```

Comment:
This is a _very_ important distinction which [we've touched on before](https://github.com/kubernetes-sigs/gateway-api/pull/993). I think it's fine to add this to the glossary as it's purely contextual, but as someone who has advocated to make this distinction within Gateway APIs in the past (coincidentally, specifically in order to support service mesh use cases with Gateway API) I can see this part in particular getting tricky. Nothing to do about it right now, but consider that the distinction of frontend vs backend with a Kubernetes `Service` may very well be something that could use its own GEP entirely.

Reply:
100% agree; the decoupling of the `Service` resource in the general Gateway API conceptual model probably makes a lot of sense. I don't think I could get behind moving away from the service resource, but clarifying the different functions of the resource is absolutely a worthwhile exercise

Reply:
Well, I can't think of a good action for the moment so I'll consider this one resolved but I definitely think this is going to be a point of great interest soon.